src/EventListener/MapToListener.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\EventListener;
+
+use AutoMapper\Attribute\MapFrom;
+use AutoMapper\Attribute\MapTo;
+use AutoMapper\Event\GenerateMapperEvent;
+use AutoMapper\Event\PropertyMetadataEvent;
+use AutoMapper\Event\SourcePropertyMetadata;
+use AutoMapper\Event\TargetPropertyMetadata;
+use AutoMapper\Exception\BadMapDefinitionException;
+use AutoMapper\Transformer\CallableTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomModelTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomModelTransformerInterface;
+use AutoMapper\Transformer\CustomTransformer\CustomPropertyTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomPropertyTransformerInterface;
+use AutoMapper\Transformer\CustomTransformer\CustomTransformersRegistry;
+use AutoMapper\Transformer\TransformerInterface;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\String\Inflector\EnglishInflector;
+use Symfony\Component\String\Inflector\InflectorInterface;
+
+final readonly class MapToListener
+{
+    public function __construct(
+        private CustomTransformersRegistry $customTransformersRegistry,
+        private InflectorInterface $inflector = new EnglishInflector(),
+    ) {
+    }
+
+    public function __invoke(GenerateMapperEvent $event): void
+    {
+        if ($event->mapperMetadata->sourceReflectionClass === null) {
+            return;
+        }
```

Comment:
here is an idea we could use all over the code to improve readability: I think we should introduce an enum which would looks like:
```php
enum MappingType
{
    case FROM_SOURCE; // (aka NORMALIZATION)
    case FROM_TARGET; // (aka DENORMALIZATION)
    case SOURCE_TARGET; // (aka OBJECTS_MAPPING)
}
```
and rely on this enum instead of relying on this kind of condition which is kinda meaningless